### ISSUE TITLE
Issues are in the incorrect order on iOS

### ISSUE THREAD

**Issue by nickoneill** [5calls/ios#253]
We just posted a new call and it appears near the bottom of the results on iOS. Previously (at least I'm pretty sure?) the order was the same as the website.

As of yet, the order is determined only by the order in which it appears in the json, we have a proper ordering value on the server side though that we could pass through to clients if it makes more sense.

<img width="383" alt="screen shot 2018-11-08 at 9 14 30 am" src="https://user-images.githubusercontent.com/49688/48215280-c5bc6f00-e336-11e8-9635-e36afb3884fd.png">

vs

![img_3541](https://user-images.githubusercontent.com/49688/48215326-dec52000-e336-11e8-8584-20d5b9bc186e.jpg)

**Comment by mccarron** [comment]
I can look tomorrow but can't today. Have we confirmed the JSON order is the same?

**Comment by nickoneill** [comment]
Yep, and the order on the website follows the JSON. For example right now the top 5:

```
      "id": "recBJ6wBlShm6b5z4",
      "name": "Demand Acting AG Matthew Whitaker Recuse Himself from Mueller Investigation",
      ...
      "id": "recFS1sCVCQhNqZUt",
      "name": "Protect Mueller and the Russia Investigation",
      ...
      "id": "recgvn5j8YZFeEytQ",
      "name": "Demand Action Against Gun Violence",
      ...
      "id": "recgUbNBp4Idrz6nk",
      "name": "Pass the Violence Against Women Reauthorization Act of 2018",
      ...
      "id": "recSQrrm6a0RR6dbK",
      "name": "Demand the Treasury Department Release Trump's Tax Returns",
      ...
```

These are mixed in the app, some at the bottom, some closer to the top of the list.

**Comment by mccarron** [comment]
Ah it's a Swift 4.2 change to how Hashes are stored, it's now randomly seeded so you can't expect the input order to be maintained. 

So we will need to sort the returned elements because now you can't count on the order of Dictionary objects.

**Comment by nickoneill** [comment]
Can we avoid storing the issues list as a dictionary? Arrays still have consistent ordering, right?

**Comment by Abizern** [comment]
Unfortunately I don't have Xcode 9 anymore so I can't load the current code. Looking at the result of the call to get issues, though it seems  we use the "issues" key which returns an array already.

**Comment by subdigital** [comment]
I'm not sure why these were stored as a dictionary to begin with.. maybe we needed some sort of lookup?  In any case if the lookup is still needed, then we can supplement with an array of ids and look them up from the dictionary.

**Comment by mccarron** [comment]
I would wager it's in the code that breaks the issues into Dictionary for use by Category functions. I would almost guarantee that the issues returned by this are now out of order in the new Dictionary this line creates. 

`let issuesByCategory = Dictionary(grouping: issues) { (issue) -> Category in`

**Comment by mccarron** [comment]
@nickoneill can you pass thru that ordering field? That would make this an easy solve. 

I dug in and the issue is, after we assemble all the issues into Categories with a Dictionary and then our view model takes all the Dictionaries's issues and makes an array of it here: 

https://github.com/5calls/ios/blob/82de7de856d2261a17366fed260dc99c4e26c623/FiveCalls/FiveCalls/IssuesViewModel.swift#L58

This is an issue now as the Category list is not the same as when it was read in due to the Swift changes. 

This would be the perfect place to sort it, but trying to get the right issues in the original order with this is not ideal since this View Model is not always for all issues. So sorting should be our best option to make sure it's always right.

**Comment by nickoneill** [comment]
Added an order field to the topic list. Note: this literally only reflects the order of the issues in the json file, so active issues may be non-contiguous (and currently are). In any case, sorting in ascending order will give you the correct order for the front page topics.

<img width="290" alt="screen shot 2018-11-20 at 9 18 13 am" src="https://user-images.githubusercontent.com/49688/48790789-fee4cf80-eca4-11e8-8c66-2afe29edbc70.png">

**Comment by nickoneill** [comment]
Fixed in #255
